src/parser/csql_grammar.y

```diff
@@ -21587,45 +21557,129 @@ opt_encrypt_algorithm
-#else
-                  $$ = 0;                        
-#endif                  
-                }
-	| '(' UNSIGNED_INTEGER ')'
-		{ DBG_TRACE_GRAMMAR(opt_deduplicate_key_level, | (LEVEL UNSIGNED_INTEGER) ); 
-                  int int_val = -1;                  
-                  if (parse_int (&int_val, $2, 10) != 0)
-		      {
-			PT_ERRORmf (this_parser, $2, MSGCAT_SET_PARSER_SYNTAX, MSGCAT_SYNTAX_INVALID_UNSIGNED_INT32, $2);
-		      }
-                  else if(int_val < DEDUPLICATE_KEY_LEVEL_MIN || int_val > DEDUPLICATE_KEY_LEVEL_MAX)
+          { DBG_TRACE_GRAMMAR(opt_index_with_clause, : );
+            container_3 ctn;
+            memset(&ctn, 0x00, sizeof(ctn));       
+            ctn.c1 = 0; // not online
+            ctn.c2 = prm_get_integer_value (PRM_ID_DEDUPLICATE_KEY_LEVEL);
+            ctn.c3 = prm_get_integer_value (PRM_ID_DEDUPLICATE_MIN_KEYS);
+            $$ = ctn; }
+        | WITH deduplicate_key_mod_level
+           { DBG_TRACE_GRAMMAR(opt_index_with_clause, | WITH deduplicate_key_mod_level );
+                container_3 ctn;
+		SET_CONTAINER_3(ctn, 0, $2, DEDUPLICATE_MIN_KEYS_UNUSE);
+		$$ = ctn;
+          }
+        ;
+
+opt_index_with_clause
+        : /* empty */
+          { DBG_TRACE_GRAMMAR(opt_index_with_clause, : );
+            container_3 ctn;
+            memset(&ctn, 0x00, sizeof(ctn));
+            ctn.c1 = 0; // not online
+            ctn.c2 = prm_get_integer_value (PRM_ID_DEDUPLICATE_KEY_LEVEL);
+            ctn.c3 = prm_get_integer_value (PRM_ID_DEDUPLICATE_MIN_KEYS);
+            $$ = ctn; }
+        | WITH index_with_item_list
+          {  DBG_TRACE_GRAMMAR(opt_index_with_clause, | WITH index_with_item_list );
+             $$ = $2;
+          DBG_PRINT}
+        ;
+
+index_with_item_list  
+        : online_parallel
+          { DBG_TRACE_GRAMMAR(index_with_item_list, : online_parallel );
+                container_3 ctn;                
+		SET_CONTAINER_3(ctn, $1, prm_get_integer_value (PRM_ID_DEDUPLICATE_KEY_LEVEL), prm_get_integer_value (PRM_ID_DEDUPLICATE_MIN_KEYS));
+		$$ = ctn;
+          }
+        | online_parallel ',' deduplicate_key_mod_level
+          { DBG_TRACE_GRAMMAR(index_with_item_list, | online_parallel ',' deduplicate_key_mod_level );
+                container_3 ctn;
+		SET_CONTAINER_3(ctn, $1, $3, DEDUPLICATE_MIN_KEYS_UNUSE);
+		$$ = ctn;
+          }
+        | deduplicate_key_mod_level
+          { DBG_TRACE_GRAMMAR(index_with_item_list, | deduplicate_key_mod_level );
+                container_3 ctn;
+		SET_CONTAINER_3(ctn, 0, $1, DEDUPLICATE_MIN_KEYS_UNUSE);
+		$$ = ctn;
+          }
+        | deduplicate_key_mod_level ',' online_parallel
+          { DBG_TRACE_GRAMMAR(index_with_item_list, | deduplicate_key_mod_level ',' online_parallel ); 
+                container_3 ctn;
+		SET_CONTAINER_3(ctn, $3, $1, DEDUPLICATE_MIN_KEYS_UNUSE);
+		$$ = ctn;
+          }
+        ;        
```

Comment:
Is there any more elegant way?
```
index_with_item_list  
        : online_parallel
        | online_parallel ',' deduplicate_key_mod_level
        | deduplicate_key_mod_level
        | deduplicate_key_mod_level ',' online_parallel
        ;
```
Even if we add only one new option, it should change like below.
```
index_with_item_list  
        : online_parallel
        | online_parallel ',' deduplicate_key_mod_level
        | online_parallel ',' <new_option>
        | online_parallel ',' deduplicate_key_mod_level ',' <new_option>
        | online_parallel ',' <new_option> ',' deduplicate_key_mod_level
        | deduplicate_key_mod_level
        | deduplicate_key_mod_level ',' online_parallel
        | deduplicate_key_mod_level ',' <new_option>
        | deduplicate_key_mod_level ',' online_parallel ',' <new_option>
        | deduplicate_key_mod_level ',' <new_option> ',' online_parallel
        | <new_option>
        | <new_option> ',' online_parallel
        | <new_option> ',' deduplicate_key_mod_level
        | <new_option> ',' online_parallel ',' deduplicate_key_mod_level
        | <new_option> ',' deduplicate_key_mod_level ',' online_parallel
        ;
```
The following code is used in postgresql.
```
alterfunc_opt_list:
                        /* At least one option must be specified */
                        common_func_opt_item
                        | alterfunc_opt_list common_func_opt_item 
                ;

common_func_opt_item:
                        CALLED ON NULL_P INPUT_P
                        | RETURNS NULL_P ON NULL_P INPUT_P
                        | STRICT_P
                        | IMMUTABLE
                        | STABLE
                        | VOLATILE
                        | EXTERNAL SECURITY DEFINER
                        | EXTERNAL SECURITY INVOKER
                        | SECURITY DEFINER
                        | SECURITY INVOKER
                        | LEAKPROOF
                        | NOT LEAKPROOF
                        | COST NumericOnly
                        | ROWS NumericOnly
                        | SUPPORT any_name
                        | FunctionSetResetClause
                        | PARALLEL ColId
                ;

static bool
compute_common_attribute(...)
{
        if (strcmp(defel->defname, "volatility") == 0)
        {
                if (is_procedure)
                        goto procedure_error;
                if (*volatility_item)
                        goto duplicate_error;

                *volatility_item = defel;
        }
        ...
duplicate_error:
        ereport(ERROR,
                        (errcode(ERRCODE_SYNTAX_ERROR),
                         errmsg("conflicting or redundant options"),
                         parser_errposition(pstate, defel->location)));
        return false;
...
```
```

Reply:
Modifications to be flexible will be carried out as a separate work later.